components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -604,6 +615,65 @@ protected String getSubjectFromUserIDClaimURI(AuthenticationContext context) {
         return subject;
     }
 
+    protected String getSubjectFromUserIDClaimURI(AuthenticationContext context, Map<String, Object> idTokenClaims)
+            throws AuthenticationFailedException {
+
+        boolean useLocalClaimDialect = context.getExternalIdP().useDefaultLocalIdpDialect();
+        String userIdClaimUri = context.getExternalIdP().getUserIdClaimUri();
```

Comment:
Given the claim body is this

{
    "userIdClaim": {
        "uri": "sub"
    },
    "roleClaim": {
        "uri": "role"
    },
    "mappings": [
        {
            "idpClaim": "role",
            "localClaim": {
                "id": "aHR0cDovL3dzbzIub3JnL2NsYWltcy9yb2xlcw",
                "uri": "http://wso2.org/claims/roles",
                "displayName": "Roles"
            }
        },
        {
            "idpClaim": "sub",
            "localClaim": {
                "id": "aHR0cDovL3dzbzIub3JnL2NsYWltcy9leHRlcm5hbGlk",
                "uri": "http://wso2.org/claims/externalid",
                "displayName": "External ID"
            }
        },
        {
            "idpClaim": "email",
            "localClaim": {
                "id": "aHR0cDovL3dzbzIub3JnL2NsYWltcy91c2VybmFtZQ",
                "uri": "http://wso2.org/claims/username",
                "displayName": "Username"
            }
        }
    ],
    "provisioningClaims": []
}

why do we iterate through mappings, instead of below

idTokenClaims.get(userIdClaimUri);